Respect umask settings in Transmission config

anderslundstedt opened the pull request:
Accidentally closed old pull request #10050 when trying to rebase it. Here is a new one.

In NixOS/nixpkgs@da6bc44 @thoughtpolice
made the Transmission NixOS module override the umask setting in the
Transmission config. This commit removes that override.

I want a different umask setting and I guess it is possible that other
people might want it to. Thus I think it is a good idea to respect the
umask settings in the Transmission config.

joachifm commented:
It occurred to me: one way to retain the old behavior but still respect the user would be to switch the original statement around, so that the user config overrides the module instead of the other way around wrt. umask. Something like
```nix
fullSettings = { 
  umask = 2; download-dir = downloadDir; incomplete-dir = incompleteDir;
} // cfg.settings
```
What do you think? This would mean that people who don't care to set the umask get the old behavior (transmission defaults to 18 aka 022 in octal).

anderslundstedt committed:
Respect umask settings in Transmission config

In NixOS/nixpkgs@da6bc44 @thoughtpolice
made the Transmission NixOS module override the umask setting in the
Transmission config. This commit removes that override.

I want a different umask setting and I guess it is possible that other
people might want it to. Thus I think it is a good idea to respect the
umask settings in the Transmission config.

anderslundstedt commented:
Looks good. Just updated the pull request.

joachifm commented:
Great, thanks
